hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/KeyDeletingService.java

```diff
@@ -220,8 +225,15 @@ public BackgroundTaskResult call() {
           if (keyBlocksList != null && !keyBlocksList.isEmpty()) {
             delCount = processKeyDeletes(keyBlocksList,
                 getOzoneManager().getKeyManager(),
-                pendingKeysDeletion.getKeysToModify(), null, expectedPreviousSnapshotId);
+                pendingKeysDeletion.getKeysToModify(), null, expectedPreviousSnapshotId, true);
             deletedKeyCount.addAndGet(delCount);
+            metrics.setIterationKeyRunCount(run);
+            metrics.setIterationKeyStartTime(startTime);
+            metrics.setIterationKeyDuration(Time.monotonicNow() - startTime);
+            metrics.setIterationKeysProcessed(keyBlocksList.size());
+            metrics.setIterationKeysSentForPurge(delCount);
+            metrics.incrNumKeysProcessed(keyBlocksList.size());
```

Comment:
Check for naming, blocks and keys separately.

Reply:
KeyBlockList has the list of `BlockGroup` objects, where each `BlockGroup` object has (key, blocklist for the key). Hence the `keyBlocksList.size()` would give the number of keys that were handled, that is why I have named it as key

Reply:
`NumKeysProcessed` is count of all keys checked (having both type of results from scm), but `NumKeysSentForPurge` is having only the successfully deleted count, they are different